File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.java
class Solution {
    public int[] rearrangeArray(int[] nums) {
        int n=nums.length;
        int pos=0;
        int neg=1;
        int[] ans=new int[n];
        for(int it:nums){
            if(it>0){
                ans[pos]=it;
                pos+=2;
            }else{
                ans[neg]=it;
                neg+=2;
            }
        }
        return ans;
    }
}

File: 2271-rearrange-array-elements-by-sign/rearrange-array-elements-by-sign.cpp
class Solution {
public:
    vector<int> rearrangeArray(vector<int>& nums) {
        int pos = 0;
        int neg = 1;
        int n = nums.size();
        vector<int> ans(n);
        for (auto& it : nums) {
            if (it > 0) {
                ans[pos] = it;
                pos += 2;
            } else {
                ans[neg] = it;
                neg += 2;
            }
        }
        return ans;
    }
};